sn_node/src/put_validation.rs

```diff
@@ -476,18 +491,42 @@ impl Node {
             .reward_wallet_balance
             .set(wallet.balance().as_nano() as i64);
 
-        // check payment is sufficient
-        let current_store_cost =
-            self.network.get_local_storecost().await.map_err(|e| {
-                ProtocolError::RecordNotStored(pretty_key.clone(), format!("{e:?}"))
-            })?;
+        if royalties_cash_notes.is_empty() {
+            return Err(ProtocolError::NoNetworkRoyaltiesPayment(pretty_key.clone()));
+        }
+
+        // publish a notification over gossipsub topic TRANSFER_NOTIF_TOPIC for the network royalties payment.
+        match bincode::serialize(&royalties_cash_notes) {
+            Ok(serialised) => {
+                let royalties_pk = *NETWORK_ROYALTIES_PK;
+                let topic = TRANSFER_NOTIF_TOPIC.to_string();
+                let mut msg: Vec<u8> = royalties_pk.to_bytes().to_vec();
+                msg.extend(serialised);
+                if let Err(err) = self.network.publish_on_topic(topic.clone(), msg) {
+                    debug!("Failed to publish a network royalties payment notification over gossipsub for record {pretty_key} and beneficiary {royalties_pk:?}: {err:?}");
+                }
+            }
+            Err(err) => debug!("Failed to serialise network royalties payment data to publish a notification over gossipsub for record {pretty_key}: {err:?}"),
```

Comment:
just to confirm, with both of the above two error cases,
it is still OK to carry on to the `store cost calculate and upload process`, right?

Reply:
I'd think so, the payment for the node will still be ok if the amount is correct